presto-main/src/main/java/com/facebook/presto/server/remotetask/ContinuousBatchTaskStatusFetcher.java

```diff
@@ -0,0 +1,146 @@
+
+    public ContinuousBatchTaskStatusFetcher(
+            Consumer<Throwable> onFail,
+            Duration refreshMaxWait,
+            Executor executor,
+            HttpClient httpClient,
+            Duration maxErrorDuration,
+            ScheduledExecutorService errorScheduledExecutor,
+            RemoteTaskStats stats,
+            boolean isBinaryTransportEnabled) {
+        idWorkerMap = new ConcurrentHashMap<>();
+        workerTaskMap = new ConcurrentHashMap<>();
+
+        this.onFail = requireNonNull(onFail, "onFail is null");
+
+        this.refreshMaxWait = requireNonNull(refreshMaxWait, "refreshMaxWait is null");
+
+        this.executor = requireNonNull(executor, "executor is null");
+        this.httpClient = requireNonNull(httpClient, "httpClient is null");
+
+        this.errorTracker = new RequestErrorTracker(taskId, initialTaskStatus.getSelf(), maxErrorDuration, errorScheduledExecutor, "getting task status"); //TODO: Fix
+        this.stats = requireNonNull(stats, "stats is null");
+        this.isBinaryTransportEnabled = isBinaryTransportEnabled;
+    }
+
+    public void addTask(
+            TaskId taskId,
+            TaskStatus initialTaskStatus,
+            Codec<List<TaskStatus>> taskListStatusCodec)
+    {
+        Task newTask = new Task();
+        requireNonNull(initialTaskStatus, "initialTaskStatus is null");
+
+        newTask.taskId = requireNonNull(taskId, "taskId is null");
+        newTask.taskStatus = new StateMachine<>("task-" + taskId, executor, initialTaskStatus);
+        newTask.taskListStatusCodec = requireNonNull(taskListStatusCodec, "taskListStatusCodec is null")
+
+        String worker = initialTaskStatus.getSelf().getHost();
+        idWorkerMap.put(taskId, worker);
+        if (!workerTaskMap.containsKey(worker)) {
+            workerTaskMap.put(worker, new WorkerTaskStatusFetcher(worker, onFail, refreshMaxWait, taskListStatusCodec, executor, httpClient, stats, isBinaryTransportEnabled));
+        }
+        workerTaskMap.get(worker).addTask(newTask);
+    }
+
+    @PostConstruct
+    public synchronized void start()
+    {
+        if (isRunning()) { // We have already called start
+            return;
+        }
+        scheduleNextRequest();
+    }
+
+    public synchronized void stop()
+    {
+        if (future != null) {
+            // do not terminate if the request is already running to avoid closing pooled connections
+            future.cancel(false);
+            future = null;
+        }
+    }
+
+    private synchronized void scheduleNextRequest()
+    {
+        for (WorkerTaskStatusFetcher workerTaskStatusFetcher: workerTaskMap.values()) {
+            workerTaskStatusFetcher.scheduleNextRequest();
+        }
+    }
+
+    TaskStatus getTaskStatus(TaskId taskId)
+    {
+        WorkerTaskStatusFetcher worker = workerTaskMap.get(idWorkerMap.get(taskId));
+        return worker.getTaskStatus(taskId);
+    }
+
+    public synchronized boolean isRunning()
```

Comment:
i believe this method is safe to remove as it's not going to be referred anywhere.